sniffles/test/test_rule_reader.py

```diff
@@ -4,6 +4,37 @@
 
 class TestRuleReader(TestCase):
 
+    def test_background_traffic_rule(self):
+
+        myrule = BackgroundTrafficRule()
+        protocols = ['http', 'ftp', 'pop', 'smtp', 'imap']
+
+        # Asserts for rule settings
+        self.assertEqual(myrule.getProto(), 'tcp')
+        for protocolType in protocols:
+            myrule = BackgroundTrafficRule(protocolType)
+            self.assertEqual(myrule.getProtocolType(), protocolType)
+            if myrule.getProtocolType() == 'http':
+                self.assertEqual(myrule.getDport(), '80')
```

Comment:
Just a suggestion, you can use a list of ports for any particular prototcol.  In fact, if you look in the ruletrafficgenerator, you will find port lists for all kinds of protocols like http, ftp, etc.